TESTS/mbed_drivers/timeout/timeout_tests.h

```diff
@@ -0,0 +1,416 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_TIMEOUT_TESTS_H
+#define MBED_TIMEOUT_TESTS_H
+
+#include "mbed.h"
+#include "unity/unity.h"
+#include "rtos/rtos.h"
+
+#define NUM_TIMEOUTS 16
+#define DRIFT_TEST_PERIOD_US 10000
+
+const float TEST_DELAY_S = 0.01;
+const uint32_t TEST_DELAY_MS = 1000.0F * TEST_DELAY_S;
+const us_timestamp_t TEST_DELAY_US = 1000000.0F * TEST_DELAY_S;
+
+/* Timeouts are quite arbitrary due to large number of boards with varying level of accuracy */
+#define LONG_DELTA_US (100000)
+#define SHORT_DELTA_US (600)
```

Comment:
not enough for nrf51_dk

Reply:
Increased `SHORT_DELTA_US` to 1200 us and updated test case delay values.